src/middlewares/logger.middleware.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable, Logger, NestMiddleware } from '@nestjs/common';
+import { Request, Response, NextFunction } from 'express';
+import { JwtService } from '@nestjs/jwt';
+
+@Injectable()
+export class LoggerMiddleware implements NestMiddleware {
+  private readonly logger = new Logger();
+  private readonly jwt = new JwtService();
+
+  use(req: Request, res: Response, next: NextFunction) {
+    const token = req.headers?.authorization?.split(' ')[1] || '';
+    const userDetail = this.jwt.decode(token) || {};
+
+    this.logger.debug(
+      `Request Method:[${req.method}] || Request URI:${
+        req.baseUrl
+      } || Request Body:[${JSON.stringify(
+        req.body,
+      )}] || Request Query Params:[${JSON.stringify(req.query)}] Request IP:[${
+        req.ip
+      }] || User Details: [${JSON.stringify(userDetail)}]`,
+    );
+    next();
```

Comment:
PullProofAI

Lack of error handling for decoding JWT token and potential exceptions.
- const userDetail = this.jwt.decode(token) || {};
+ let userDetail = {};
+ try {
+   userDetail = this.jwt.decode(token) || {};
+ } catch (error) {
+   this.logger.error(`Error decoding JWT token: ${error.message}`);
+ }
```

#### Explanation:
Adding a try-catch block around the JWT token decoding ensures that any errors during decoding are caught and logged appropriately. This prevents unhandled exceptions that could disrupt the application flow and provides better visibility into potential issues with token decoding.


<!-- This is an auto-generated comment by PullProofAI -->